### PR TITLE
[google_maps_flutter][iOS 17] takeSnapshot FIX

### PR DESCRIPTION
**Pull request by GavrielRosendorn74**
I have fixed the function takeSnapShot for IOS17.

Here you got some screenshots of before now : 

Before : 

289014755-4e6787c3-c7c0-454a-9c3f-89730a846302

Now : 

289014438-4bc5a965-08e0-4520-984d-e5c054c1f370

This issue was fixed by this PR :[ #139733 ](https://github.com/flutter/flutter/issues/139733)

## Pre-launch Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md:3]: can you be more descriptive here

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:256]: can you explain why this fixes the issue? also have you verified earlier ios version to make sure it still works?

Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:256]: Hi @hellohuanlin ! This is actually suppose to do the same thing. The method i now use is alternative to the other one. I have made test on IOS 17 and also earlier version it look like it's working well :)

File: packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md
Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md]: done :)

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (hellohuanlin) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:256]: i may miss something here - is it now doing different things (one fixes the issue and the other does not?) if that's the case, can you add some comments doucmenting this behavior?

**Comment by GavrielRosendorn74** [comment]
Now i just changed 2.3.5 to 2.36 for the package version.

**Comment by GavrielRosendorn74** [comment]
Now i just changed 2.3.5 to 2.36 for the package version.

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (hellohuanlin) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:256]: Can you add a comment there saying the other API originally used doesn't work in ios 17? (even if you don't know why it doesn't work).

**Review by hellohuanlin — COMMENTED** [review]
Thanks for contributing. Can you add the comment as we discussed in my previous review?

**Comment by GavrielRosendorn74** [comment]
Sorry for being inactive here for weeks. I am solving all these points today :)

File: packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md
Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md]: done

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:257]: done

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:255]: can you provide more details other than just "doesn't work"? Also please explicitly put the original API here (we can't tell what "The API originally used" is after you remove it.

Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m]: Yep ! Is it enough right now ?

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (hellohuanlin) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:260]: Please don't put the whole chunk of code. How about this: 

> For some unknown reason `mapView.layer::renderInContext` API returns a blank image on iOS 17. So we have to use `drawViewHierarchyInRect` API.

Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:260]: Let's use this ! :)

**Review by hellohuanlin — APPROVED** [review]

**Comment by GavrielRosendorn74** [comment]
@stuartmorgan you're the only one can merge or is it automatic ?

**Comment by stuartmorgan** [comment]
@hellohuanlin When you approve a PR that requires two reviews, please add a second reviewer.

**Comment by GavrielRosendorn74** [comment]
@stuartmorgan thanks for adding yourself as reviewer ! ;)

**Comment by GavrielRosendorn74** [comment]
Hi @stuartmorgan  ! How are you ? Is there some changes to add to this PR ?

**Comment by stuartmorgan** [comment]
> Hi @stuartmorgan ! How are you ? Is there some changes to add to this PR ?

I won't know that until I have reviewed it, which I have not done in the one working day I've had since I marked it for review. Please see https://github.com/flutter/flutter/wiki/Tree-hygiene for expectations around review turnaround time.

**Comment by GavrielRosendorn74** [comment]
> > Hi @stuartmorgan ! How are you ? Is there some changes to add to this PR ?
> 
> I won't know that until I have reviewed it, which I have not done in the one working day I've had since I marked it for review. Please see https://github.com/flutter/flutter/wiki/Tree-hygiene for expectations around review turnaround time.

Got it ! Thanks ! :)

File: packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md:3]: The function is called `takeSnapshot`, not `takeSnapShot` (and should also be enclosed in backticks, since it's a code reference.

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m:260]: This can be inlined into the `result` call; having the separate variable just duplicates the type, which makes this less readable IMO.

File: packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md
Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md]: done.

Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md]: done.

Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/CHANGELOG.md]: done.

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m
Comment (GavrielRosendorn74) [reply packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m]: done.

**Comment by GavrielRosendorn74** [comment]
Hi @stuartmorgan. All the fixes you have asked for are done :)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
The code LGTM, but the CI failures here need to be fixed before this can land.

**Comment by GavrielRosendorn74** [comment]
@stuartmorgan that's it i think on this time ! 🚀

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**
